### PR TITLE
os/tools/compression: Add gitignore for compression folder

### PR DESCRIPTION
**Pull request by sijupunnnoose**
Compression tool generates temp files and executable for
binary compression which need not to be added to git.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [214d734227f5ec386e7eb86bb176be08ebaebd35] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [214d734227f5ec386e7eb86bb176be08ebaebd35] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [31af83f80fcaf28fe9fbd2509fdc87060a0404e3] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [31af83f80fcaf28fe9fbd2509fdc87060a0404e3] -

**Comment by sunghan-chang** [comment]
Could you check all of copied or linked or generated files are deleted with clean command?

**Comment by sijupunnnoose** [comment]
> Could you check all of copied or linked or generated files are deleted with clean command?

.config was missing in clean, add it in clean

**Comment by seinfra** [comment]
Target : [703f0db21427085887bd760aebe6557d6d435ce9] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [703f0db21427085887bd760aebe6557d6d435ce9] - Code Rule Check OK.

**Comment by sunghan-chang** [comment]
@sijupunnnoose
1. `distclean` target is missing either.
Because of that, it shows below:
```
make[1]: Entering directory '/home/sunghan/Work/TinyAra/Forked_TizenRT/os/tools/compression'
make[1]: *** No rule to make target 'distclean'.  Stop.
```
2. When I call clean, only tools/compression shows unnecessary information
```
make[1]: Entering directory '/home/sunghan/Work/TinyAra/Forked_TizenRT/os/tools/compression'
CP: .config
=== cleaning ===
make[1]: Leaving directory '/home/sunghan/Work/TinyAra/Forked_TizenRT/os/tools/compression'
```
Please remove them.
3. Why does it call `CP` with clean command?

**Comment by sunghan-chang** [comment]
4. could you split commits for clean and distclean from gitignore commit?

**Comment by seinfra** [comment]
Target : [b50d72cb3e01f863d8b8b690229603410d404103] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [b50d72cb3e01f863d8b8b690229603410d404103] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [1392ad0e5a302109fa3168b16555a9657132ca4e] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [1392ad0e5a302109fa3168b16555a9657132ca4e] - Code Rule Check (C++) OK.

**Comment by sijupunnnoose** [comment]
> @sijupunnnoose
> 
> 1. `distclean` target is missing either.
>    Because of that, it shows below:
> 
> ```
> make[1]: Entering directory '/home/sunghan/Work/TinyAra/Forked_TizenRT/os/tools/compression'
> make[1]: *** No rule to make target 'distclean'.  Stop.
> ```
> 
> 1. When I call clean, only tools/compression shows unnecessary information
> 
> ```
> make[1]: Entering directory '/home/sunghan/Work/TinyAra/Forked_TizenRT/os/tools/compression'
> CP: .config
> === cleaning ===
> make[1]: Leaving directory '/home/sunghan/Work/TinyAra/Forked_TizenRT/os/tools/compression'
> ```
> 
> Please remove them.
> 3. Why does it call `CP` with clean command?

Added distclean section and removed unnecessary information

**Comment by sijupunnnoose** [comment]
> 1. could you split commits for clean and distclean from gitignore commit?

Done

File: os/tools/compression/Makefile [hunk 1]
```diff
@@ -90,29 +90,25 @@ $(APPNAME): Makefile $(CONFIG) $(CCONFIG) $(COMPHEADER) $(OBJECTS)
 	@$(CC) $(LDFLAGS) $(OBJECTS) $(LIBFILES) -o $@
```
Comment (sunghan-chang) [inline os/tools/compression/Makefile]: Is this CC? not LD?

File: os/tools/compression/Makefile [hunk 2]
```diff
@@ -90,29 +90,25 @@ $(APPNAME): Makefile $(CONFIG) $(CCONFIG) $(COMPHEADER) $(OBJECTS)
...
 $(COMPHEADER): $(TINYARADIR)/include/tinyara/binfmt/compression/compression.h
-	@echo "CP: compression.h"
 	@cp $(TINYARADIR)/include/tinyara/binfmt/compression/compression.h .
 
 # =================================
 # Rule to retrieve the .config file
 # =================================
 $(CONFIG): $(TINYARADIR)/.config
-	@echo "CP: .config"
 	@cp $(TINYARADIR)/.config .
```
Comment (sunghan-chang) [inline os/tools/compression/Makefile]: As I left before, when we call `clean`, this is executed. Could you check the reason of calling this with clean?

File: os/tools/compression/Makefile [hunk 3]
```diff
@@ -122,3 +118,9 @@ endif
 	@rm -f config.h
 	@rm -f compression.h
 	@rm -f *.h
+	@rm -f .config
+
+distclean: clean
+	$(call DELFILE, Make.dep)
+	$(call DELFILE, .depend)
```
Comment (sunghan-chang) [inline os/tools/compression/Makefile]: To use definition like DELFILE, you must include `Config.mk` by including Make.defs of os folder. But here is no including. Is this really working?

File: os/tools/compression/Makefile [hunk 4]
```diff
@@ -122,3 +118,9 @@ endif
 	@rm -f config.h
 	@rm -f compression.h
 	@rm -f *.h
+	@rm -f .config
```
Comment (sunghan-chang) [inline os/tools/compression/Makefile]: Let's use DEFFILE and DELDIR

File: os/tools/compression/Makefile (hunk 1 above)
Comment (sijupunnnoose) [reply os/tools/compression/Makefile]: It is CC , we are building the tool for compressing image for ubuntu machine. Not cross compiling

File: os/tools/compression/Makefile (hunk 3 above)
Comment (sijupunnnoose) [reply os/tools/compression/Makefile]: I missed to -include $(TOPDIR)/Make.defs, will correct it

File: os/tools/compression/Makefile (hunk 4 above)
Comment (sijupunnnoose) [reply os/tools/compression/Makefile]: Will change to DELFILE &DELDIR

File: os/tools/compression/Makefile (hunk 2 above)
Comment (sijupunnnoose) [reply os/tools/compression/Makefile]: It is not calling copy now with clean.

**Comment by seinfra** [comment]
Target : [436364ff773edafa8fb00a9ae4689e3c570b6d2d] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [436364ff773edafa8fb00a9ae4689e3c570b6d2d] - Code Rule Check OK.

**Pull request merged**